hs/Database/CourseInsertion.hs

```diff
@@ -11,16 +11,46 @@ into the database. These functions are used as helpers for the WebParsing module
 module Database.CourseInsertion
     (insertCourse,
      setTutorialEnrolment,
-     setPracticalEnrolment) where
+     setPracticalEnrolment,
+     saveGraphJSON,
+     insertGraph) where
 
 import qualified Data.Text as T
-
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import Happstack.Server.SimpleHTTP
 import Control.Monad.IO.Class (MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Data.Maybe (fromMaybe)
-
-import Database.Persist.Sqlite (insert_, SqlBackend, (=.), (==.), updateWhere)
+import Config (databasePath)
+import Util.Happstack (createJSONResponse)
+import Database.Persist.Sqlite (selectFirst, fromSqlKey, update, insertMany_, insert_, insert, SqlBackend, (=.), (==.), updateWhere, runSqlite)
 import Database.Tables
+import Data.Aeson
+
+-- | Insert SVG graph and return its id
+insertGraph :: String -> IO Response
+insertGraph nameStr = runSqlite databasePath $ do
+    nameExists <- selectFirst [GraphTitle ==. nameStr] []
+    case nameExists of
+      Nothing -> do
+        gId <- insert $ Graph nameStr 256 256
+        return $ createJSONResponse gId
+      Just _ -> do
+        gId <- insert $ Graph "" 256 256
+        update gId [GraphTitle =. ((++) "User generated graph " $ show $ fromSqlKey gId)]
```

Comment:
For now, let's not insert anything in the case that the graph name is already taken.